js/layout-javascript/news-feed-code.js

```diff
@@ -1479,6 +1479,20 @@ DynamicList.prototype.renderLoopHTML = function (iterateeCb) {
         requestAnimationFrame(render);
       } else {
         _this.$container.find('.new-news-feed-list-container').removeClass('loading').addClass('ready');
+
+        // SuperClamp works only when elemet has a fixed max-height or height
+        // To do so we need to take computed line-height and multiple it on data-line-clamp value
+        _this.Utils.Page.setClampHeight('[data-line-clamp]')
```

Comment:
All of this

```js
        _this.Utils.Page.setClampHeight('[data-line-clamp]')
          .then(function (elements) {
            Superclamp.register(elements);
            return Promise.resolve(elements);
          })
          .then(function (elements) {
            _this.Utils.Page.removeClampHeight(elements);
          })
          .catch(function (selector) {
            console.error('Fail to clamp line in the selector ' + selector);
          });
```

...seems to be the same. Why not just create a `_this.Utils.Page.setLineClamps()` function that returns a promise?